src/Network/Minio/API.hs

```diff
@@ -104,20 +106,38 @@ buildRequest ri = do
                    return $ Just $ connectRegion ci
                | otherwise -> discoverRegion ri
 
+  regionHost <-
+    case region of
+      Nothing -> do
+        return $ connectHost ci
+      Just r -> do
+        if "amazonaws.com" `T.isSuffixOf` (connectHost ci)
+          then do
+          maybe
+               (throwM $ ME $ ValidationError $ MErrVRegionNotSupported r)
+               return
+               (Map.lookup r awsRegionMap)
+          else do
+          return $ connectHost ci
```

Comment:
All the 4 `do` terms are extraneous as they contain just one statement inside. The `case region of` should also begin on the same line as `regionHost`.